### PR TITLE
Empty cart creation added

### PR DESCRIPTION
**Pull request by rogyar**
### Description
Implementation of mutation for empty cart creation by the guest/customer user.
`createEmptyCart` mutation is declared which accepts no arguments. It returns a hashed cart ID for guest customers and the ~~real~~ hashed quote ID for logged in customers. The customer status (logged in/guest) is performed by checking authorization headers of the request.

### Fixed Issues (if relevant)

1. https://github.com/magento/graphql-ce/issues/118 : [Mutations] Cart Operations > Create Empty Cart

### Manual testing scenarios
Use the following query to create an empty shopping cart:

```
mutation {
  createEmptyCart
}
```

To test with logged in customer, obtain the customer token and provide it in the request headers.

### CODE REVIEW

**Review by naydav — DISMISSED** [review]
Looks like we need to add new module to
`composer.json` (and regenerate `composer.lock`)

**Comment by paliarush** [comment]
@rogyar I have modified requirements slightly. We need to return cart ID hash for registered customers as well. Please let me know if you could do that change in current task, otherwise we will create a separate one.

Sorry for the change, but this decision was made after the task was created.

**Review by paliarush — APPROVED** [review]

File: app/code/Magento/QuoteGraphQl/Model/Resolver/Cart/CreateEmptyCart.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        $this->valueFactory = $valueFactory;
+        $this->quoteIdMaskFactory = $quoteIdMaskFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    {
+        $customerId = $context->getUserId();
```
Comment (paliarush) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/Cart/CreateEmptyCart.php]: We should avoid working with global `$context` object, it is possible to inject `UserContextInterface` via constructor instead.

Comment (rogyar) [reply app/code/Magento/QuoteGraphQl/Model/Resolver/Cart/CreateEmptyCart.php]: That was my previous approach. But I saw that we already have `UserContextInterface` within a scope of $context and it doesn't make sense to inject it once again. 
Thank you for the tip, I will roll back the changes

**Review by paliarush — APPROVED** [review]

**Review by paliarush — APPROVED** [review]

**Pull request merged**